File: example/src/main/java/ru/virgil/example/box/BoxRepository.java
package ru.virgil.example.box;

import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import ru.virgil.example.truck.Truck;
import ru.virgil.example.user.UserDetails;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

@Repository
public interface BoxRepository extends JpaRepository<Box, UUID> {

    Optional<Box> findByOwnerAndUuid(UserDetails owner, UUID uuid);

    List<Box> findAllByOwner(UserDetails owner, Pageable pageable);

    List<Box> findAllByOwner(UserDetails owner);

    List<Box> findAllByOwnerAndTruck(UserDetails owner, Truck truck, Pageable pageable);

    long countAllByOwner(UserDetails owner);

}

File: example/src/main/java/ru/virgil/example/box/BoxRepository.kt
package ru.virgil.example.box

import org.springframework.data.domain.Pageable
import org.springframework.stereotype.Repository
import ru.virgil.example.system.entity.OwnedEntityRepository
import ru.virgil.example.truck.Truck
import ru.virgil.example.user.UserDetails
import java.util.*

@Repository
interface BoxRepository : OwnedEntityRepository<Box> {

    @Deprecated("Use Deleted False")
    fun findByOwnerAndUuid(owner: UserDetails, uuid: UUID): Optional<Box>
    fun findByOwnerAndUuidAndDeletedIsFalse(owner: UserDetails, uuid: UUID): Optional<Box>

    @Deprecated("Use Deleted False")
    fun findAllByOwner(owner: UserDetails, pageable: Pageable): List<Box>
    fun findAllByOwnerAndDeletedIsFalse(owner: UserDetails, pageable: Pageable): List<Box>

    @Deprecated("Use Deleted False")
    fun findAllByOwnerAndTruck(owner: UserDetails, truck: Truck, pageable: Pageable): List<Box>
    fun findAllByOwnerAndTruckAndDeletedIsFalse(owner: UserDetails, truck: Truck, pageable: Pageable): List<Box>

    @Deprecated("Use Deleted False")
    fun countAllByOwner(owner: UserDetails): Long
    fun countAllByOwnerAndDeletedIsFalse(owner: UserDetails): Long

    @Deprecated("Use Deleted False")
    fun findAllByOwnerAndType(owner: UserDetails, boxType: BoxType): List<Box>
    fun findAllByOwnerAndTypeAndDeletedIsFalse(owner: UserDetails, boxType: BoxType): List<Box>
}
